gigascience/gigadb-website
# Filter empty keywords from keywords list on dataset view

# Pull request for issue: #2027

This is a pull request for the following functionalities:

* filter out empty keywords in the view to prevent potentialy showing "keywords" title with no keywords

this is based on guessing what the issue might be with the dataset shown in the issue as I do not have access to it in the test site

## How to test?

I don't know of a test dataset which has the same issue indicated in the ticket, if the guess is correct, ie empty string keyword, you might go to protected/views/dataset/view.php line 97 and replace the `protected/views/dataset/view.php` with `['', ' ', '   ']` and check for any dataset whether "keyword" is displayed under the description

## How have functionalities been implemented?

- i dont have access to the dataset in the ticket, which shows "keywords" followed by empty space
- I am assuming a possible cause is the presence of empty string keywords, e.g. ''
- i added a filter that filters out empty keywords from the keywords array
note: if the diagnostic is correct, then it would be also advisable to disallow empty keywords client side to begin with from wherever the user inputs them, or / and filter them out server side

## Any issues with implementation?

I do not know how to access the dataset in the ticket to better check this
